### ISSUE TITLE
Should we have a separate `circuits` DataFrame per buildilng?

### ISSUE THREAD

**Issue by JackKelly** [nilmtk/nilmtk#64]
In issue #12 ("How to represent a Building in memory?") I had argued:

> If we store the electrical wiring configuration in a graph data structure (which allows us to naturally represent arbitrary levels of meter hierarchies) and given that there isn't always a perfectly clean distinction between what's a "circuit" and what's an "appliance" then I wonder if we can simplify our Electricity class by removing the "circuits" DataFrame and renaming the "appliances" DataFrame to something like "submetered"? Then both circuits and appliance data would go into the" submetered" DataFrame and our "wiring" graph can represent arbitrary levels of meter hierarchies whilst reducing the code complexity. What do you think? It would be lovely if our code could easily work with any metering setup, from simple domestic to large commercial systems, without having to change any code ;) 

After some further discussion, we decided to lean towards keeping `circuits` and `appliances` separate.

However, after having spent some more time thinking about importing the REDD dataset, I am starting to lean heavily back towards the idea of using a single `submetered` dataframe both both appliances and circuits.  The reason is simple:  in REDD, the labels do not give much information about whether a channel is a single appliance or a circuit (I believe that most are circuits?).  Given this ambiguity, I'd feel much more comfortable doing something like this:

`Building` would have just two dataframes: `mains` and `submetered`.  All circuit and appliance data would go into `submetered`.  Hence, in cases like REDD where the distinction between `appliance` and `circuit` is ambiguous based on the labels, we do not have to make an uncertain decision when we import the data. Instead we just put all the submetered data into the `submetered` dataframe.  We would then have a function to automatically infer the "circuit diagram" (#29) and establish the meter hierarchy; this would be stored in our wiring graph.

To summarise: if we could always be certain of the distinction between a `circuit` and an `appliance` then I would agree that we should have separate dataframes for these.  But, given that there is significant uncertainty (based on information from the labels alone) then I think we should opt for having just a single `submetered` dataframe.

Collapsing `appliances` and `circuits` into a single `submetered` dataframe should both simplify our code and allow us to more cleanly model the world we're trying to model.  Unless I've missed something?

What do you think?
